plugins/inputs/windows_event_log/wineventlog/utils.go

```diff
@@ -99,6 +111,28 @@ func UTF16ToUTF8Bytes(in []byte, length uint32) ([]byte, error) {
 	return decoded, err
 }
 
+func IsTheEndOfContent(in []byte, length uint32) bool {
+	// scan next 100 characters, if any of them is none '0', return false
+	i := int(length)
+
+	if i%2 != 0 {
+		i -= 1
+	}
+	max := len(in)
+	if i+100 < max {
+		max = i+100
+	}
+
+	for ; i < max - 2; i += 2 {
+		v1 := uint16(in[i+2]) | uint16(in[i+1])<<8
```

Comment:
I will have to reread this again to understand it, but would be nice to have a comment that explains why we do the bitwise OR here

Reply:
It is combining two bytes(8 bits) into uint16.

Reply:
You can skip the conversion and bit shift and just do:
```
for ; i < max - 2; i += 2 {
    if in[i+1] != 0 || in[1+2] != 0 {
        return false
    }
```
Or just
```
for ; i < max; i++ {
    if in[1] != 0 {
        return false
    }
}
```